is_core/filters/default_filters.py

```diff
@@ -187,66 +221,174 @@ def get_widget(self, request):
 
 class BooleanFieldFilter(DefaultFieldFilter):
 
+    def clean_value(self, value, suffix):
+        if value not in ['0', '1']:
+            raise FilterValueException(ugettext('Value can be only "0" or "1"'))
+        return value == '1'
+
     def get_widget(self, request):
-        return forms.Select(choices=(('', '-----'), (1, _('Yes')), (0, _('No'))))
+        return forms.Select(choices=(('', '-----'), (1, ugettext('Yes')), (0, ugettext('No'))))
 
 
-class NunberFieldFilter(DefaultFieldFilter):
+class NumberFieldFilter(DefaultFieldFilter):
 
     suffixes = ['gt', 'lt', 'gte', 'lte', 'in', 'isnull']
 
 
+class IntegerNumberFieldFilter(NumberFieldFilter):
+
+    def clean_value(self, value):
+        try:
+            return int(value)
+        except ValueError:
+            raise FilterValueException(ugettext('Value "{}" must be integer.'.format(value)))
+
+
+class FloatNumberFieldFilter(NumberFieldFilter):
+
+    def clean_value(self, value):
+        try:
+            return float(value)
+        except ValueError:
+            raise FilterValueException(ugettext('Value "{}" must be float.'.format(value)))
+
+
+class DecimalNumberFieldFilter(NumberFieldFilter):
+
+    def clean_value(self, value):
+        try:
+            return Decimal(value)
+        except InvalidOperation:
+            raise FilterValueException(ugettext('Value "{}" must be decimal.'.format(value)))
+
+
 class RelatedFieldFilter(DefaultFieldFilter):
-    pass
+
+    def get_last_rel_field(self, field):
```

Comment:
tohle je dost tricky, musím zkontrolovat že má FK správnou hodnotu nicméně to potřebuju datový typ pole na který fk ukazuje, nicméně pokud mají modely dědičnost tak fk může ukazovat na fk a já musím nalézt to poslední pole které už cizím klíčem není a zkontrolovat na něm zda je value ve správném datovém typu k tomu mi pomáhá get_prep_value